src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultDocumentContextFactory.cs

```diff
@@ -100,12 +101,24 @@ private bool TryResolveDocument(string filePath, VSProjectContext? projectContex
         }
 
         var project = _projectSnapshotManagerAccessor.Instance.GetLoadedProject(projectContext.ToProjectKey());
-        if (project.GetDocument(filePath) is { } document)
+        if (project?.GetDocument(filePath) is { } document)
         {
             documentSnapshot = document;
             return true;
         }
 
+        // Couldn't find the document in a real project. Maybe the language server doesn't yet know about the project
+        // that the IDE is asking us about. In that case, we might have the document in our misc files project, and we'll
+        // move it to the real project when/if we find out about it.
+        var miscellaneousProject = _snapshotResolver.GetMiscellaneousProject();
+        var normalizedDocumentPath = FilePathNormalizer.Normalize(filePath);
+        if (miscellaneousProject.GetDocument(normalizedDocumentPath) is { } miscDocument)
```

Comment:
As noted in the commit description, this was a regression in the multi-targeting work. Same scenario as the above comment, where VS knows about the project early, but we just weren't looking up the misc files project at all. The "old" code on line 100, above, does use the misc files project, so this was just a straight oversight from me.